src/main/java/com/google/api/generator/engine/ast/ThrowExpr.java

```diff
@@ -47,22 +50,37 @@ public Builder setMessageExpr(String message) {
 
     public abstract Builder setMessageExpr(Expr expr);
 
+    public abstract Builder setCauseExpr(Expr expr);
+
     // Private.
     abstract TypeNode type();
 
     abstract Expr messageExpr();
 
+    abstract Expr causeExpr();
+
     abstract ThrowExpr autoBuild();
 
     public ThrowExpr build() {
       Preconditions.checkState(
           TypeNode.isExceptionType(type()),
           String.format("Type %s must be an exception type", type()));
+
       if (messageExpr() != null) {
         Preconditions.checkState(
             messageExpr().type().equals(TypeNode.STRING),
             String.format("Message expression type must be a string for exception %s", type()));
       }
+
+      if (causeExpr() != null) {
+        Preconditions.checkState(
+            ConcreteReference.withClazz(Throwable.class)
```

Comment:
Minor, but probably we don't have to create reference each time, and can have it as a static final constant in this class.

Reply:
Done, added to TypeNode.